src/katsdpcontroller/generator.py

```diff
@@ -208,6 +208,59 @@ def aggregate_compute(self) -> Tuple[Sensor.Status, int]:
         return (status, self._total)
 
 
+class SyncSensor(SimpleAggregateSensor[bool]):
+    """Aggregate which tracks whether its children are synchronised.
+
+    In this case,
+    - a 'synchronised' child has a reading of (True, NOMINAL), and
+    - an 'unsynchronised' child has a reading of (False, ERROR).
+    """
+
+    def __init__(
+        self, target: SensorSet, sensor_type: Type[bool],
+        name: str, description: str, units: str = "",
+        *,
+        auto_strategy: Optional["SensorSampler.Strategy"] = None,
+        auto_strategy_parameters: Iterable[Any] = (),
+        name_regex: re.Pattern,
+        children: int
+    ) -> None:
+        self.target = target
+        self.name_regex = name_regex
+        self.children = children
```

Comment:
I'd just suggest making this `n_children` just so that it's obvious that it's a count, rather than a set or list or something.
Same goes for the parameter above.